lib/actions/local/configure.coffee

```diff
@@ -42,15 +45,133 @@ CONFIGURATION_SCHEMA =
 			fileset: true
 			type: 'ini'
 			location:
-				path: 'system-connections'
-				partition:
-					primary: 1
+				path: CONNECTIONS_FOLDER.slice(1)
+				partition: BOOT_PARTITION
 		config_json:
 			type: 'json'
 			location:
 				path: 'config.json'
-				partition:
-					primary: 1
+				partition: BOOT_PARTITION
+
+inquirerOptions = (data) -> [
+	{
+		message: 'Network SSID'
+		type: 'input'
+		name: 'networkSsid'
+		default: data.networkSsid
+	}
+	{
+		message: 'Network Key'
+		type: 'input'
+		name: 'networkKey'
+		default: data.networkKey
+	}
+	{
+		message: 'Do you want to set advanced settings?'
+		type: 'confirm'
+		name: 'advancedSettings'
+		default: false
+	}
+	{
+		message: 'Device Hostname'
+		type: 'input'
+		name: 'hostname'
+		default: data.hostname,
+		when: (answers) ->
+			answers.advancedSettings
+	}
+	{
+		message: 'Do you want to enable persistent logging?'
+		type: 'confirm'
+		name: 'persistentLogging'
+		default: data.persistentLogging
+		when: (answers) ->
+			answers.advancedSettings
+	}
+]
+
+getConfiguration = (data) ->
+	_ = require('lodash')
+	inquirer = require('inquirer')
+
+	# `persistentLogging` can be `undefined`, so we want
+	# to make sure that case defaults to `false`
+	data = _.assign data,
+		persistentLogging: data.persistentLogging or false
+
+	inquirer.prompt(inquirerOptions(data))
+	.then (answers) ->
+		return _.merge(data, answers)
+
+# Taken from https://goo.gl/kr1kCt
+CONNECTION_FILE = '''
+	[connection]
+	id=resin-wifi
+	type=wifi
+
+	[wifi]
+	hidden=true
+	mode=infrastructure
+	ssid=My_Wifi_Ssid
+
+	[wifi-security]
+	auth-alg=open
+	key-mgmt=wpa-psk
+	psk=super_secret_wifi_password
+
+	[ipv4]
+	method=auto
+
+	[ipv6]
+	addr-gen-mode=stable-privacy
+	method=auto
+'''
+
+prepareConnectionFile = (target) ->
```

Comment:
This is a little complicated, and a quick summary comment above it (basically the bullet points from this PR's description, which are great) would be useful :+1: